components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIStoreHostObject.java

```diff
@@ -1737,6 +1737,131 @@ public static NativeObject jsFunction_getAllPaginatedPublishedAPIs(Context cx, S
 
     }
 
+    public static NativeObject jsFunction_getAllPaginatedPublishedLightWeightAPIs(Context cx, Scriptable thisObj,
+                                                                                  Object[] args, Function funObj)
+            throws ScriptException, APIManagementException {
+
+        APIConsumer apiConsumer = getAPIConsumer(thisObj);
+        String tenantDomain;
+        boolean returnAPItags = false;
+        boolean lightWeight = false;
+        String [] statusList = {APIConstants.PUBLISHED};
+        if (args[0] != null) {
+            tenantDomain = (String) args[0];
+        } else {
+            tenantDomain = MultitenantConstants.SUPER_TENANT_DOMAIN_NAME;
+        }
+
+        int start = Integer.parseInt((String) args[1]);
+        int end = Integer.parseInt((String) args[2]);
+
+        if (args.length > 3 && args[3] != null) {
+            returnAPItags = Boolean.parseBoolean((String) args[3]);
+        }
+        return getPaginatedLightWeightAPIsByStatus(apiConsumer, tenantDomain, start, end, statusList, returnAPItags);
+    }
+
+    private static NativeObject getPaginatedLightWeightAPIsByStatus(APIConsumer apiConsumer, String tenantDomain,
```

Comment:
Lets add java docs

Reply:
Done in  ccca5d254427d1d6d57b5c373cdf746ec51dc6e4

Reply:
Shall we define the params too without just stating them?

Reply:
Done in dfada70927ee9a5f40be8bc04d23bf3f0bd5d280